unittests/testrunnerstep.py

```diff
@@ -115,73 +120,49 @@ def test_runner_step(self):
         )
 
         self.assertEqual(
-            step.args(self.asset_id_method),
-            [],
-            msg="Incorrect args",
-        )
-        # a second time to prove memoization is working.
-        self.assertEqual(
-            step.args(self.asset_id_method),
-            [],
-            msg="Incorrect args",
-        )
-
-        self.assertEqual(
-            step.kwargs(self.asset_id_method, {}),
-            {"asset_id": ASSET_ID},
-            msg="Incorrect kwargs",
-        )
-        # a second time to prove memoization is working.
-        self.assertEqual(
-            step.kwargs(self.asset_id_method, {}),
-            {"asset_id": ASSET_ID},
-            msg="Incorrect kwargs",
-        )
-
-        self.assertEqual(
-            step.set_asset_label,
+            step.label("set", "asset"),
             False,
             msg="Incorrect set_asset_label",
         )
         # a second time to prove memoization is working.
         self.assertEqual(
-            step.set_asset_label,
+            step.label("set", "asset"),
             False,
             msg="Incorrect set_asset_label",
         )
 
         self.assertEqual(
-            step.use_asset_label,
-            True,
+            step.label("use", "asset"),
+            "add_kwarg_asset_identity",
             msg="Incorrect use_asset_label",
         )
         # a second time to prove memoization is working.
         self.assertEqual(
-            step.use_asset_label,
-            True,
+            step.label("use", "asset"),
+            "add_kwarg_asset_identity",
             msg="Incorrect use_asset_label",
         )
 
         self.assertEqual(
-            step.use_location_label,
+            step.label("use", "location"),
             False,
             msg="Incorrect use_location_label",
         )
         # a second time to prove memoization is working.
         self.assertEqual(
-            step.use_location_label,
+            step.label("use", "location"),
             False,
             msg="Incorrect use_location_label",
         )
 
         self.assertEqual(
-            step.set_location_label,
+            step.label("set", "location"),
```

Comment:
Are these changes related to the subject end point being added?

Just so I can understand the intention :)

Reply:
if you look at the changes to Runner class in archivist/runner.py you will see that the kwarsg() method has disappered and the args() method is much  simpler.

Reply:
Sounds good to me